grype/db/v6/installation/curator.go

```diff
@@ -298,7 +298,14 @@ func (c curator) update(current *db.Description) (*distribution.Archive, error)
 	if err != nil {
 		return nil, fmt.Errorf("unable to resolve vulnerability DB URL: %w", err)
 	}
-	dest, err := c.client.Download(url, filepath.Dir(c.config.DBRootDir), mon.downloadProgress.Manual)
+
+	// Ensure parent of DBRootDir exists for the download client to create a temp dir within DBRootDir
+	// This might be redundant if DBRootDir must already exist, but good for safety.
+	if err := os.MkdirAll(c.config.DBRootDir, 0o700); err != nil {
```

Comment:
I had to look up the `0o` prefix -- should we just use `0`, since most examples tend to use this form and the remainder of the app tends to use this form?

Reply:
Hmm, it looks like `0o` was introduced in Go 1.13: https://go.dev/doc/go1.13#language

`gopls` at version `v0.18.1` will re-write `0600` as `0o600` for example, though. It seems like there is momentum (admittedly with mixed opinion) towards the `0o` one. What do you think?